libdnf/conf/option_string_list.cpp

```diff
@@ -103,11 +133,24 @@ void OptionStringList::set(Priority priority, const ValueType & value) {
     }
 }
 
-void OptionStringList::set(Priority priority, const std::string & value) {
+template <typename T>
+void OptionStringContainer<T>::set(Priority priority, const std::string & value) {
     set(priority, from_string(value));
 }
 
-std::string OptionStringList::to_string(const ValueType & value) const {
+template <typename T>
+void OptionStringContainer<T>::add_item(Priority priority, const std::string & item) {
+    assert_not_locked();
+
+    if (priority >= get_priority()) {
+        test_item(item);
+        value.insert(value.end(), item);
+        set_priority(priority);
+    }
```

Comment:
I am not sure whether test of priority for append option make sense. Normally priority is used that it ensure consistent result regardless of order. In append option it will somehow mess up with results. Proposing to test priority only for `set_priority(priority);`. What do you think?

```
    test_item(item);
    value.insert(value.end(), item);
    if (priority >= get_priority()) {
        set_priority(priority);
    }
```

Reply:
I see. Now I am thinking about it and the `add_item` is available only through the API = therefore always `Runtime`, right? What about to drop the priority parameter for `add_item` and always set the `Runtime`?